mayastor/src/subsys/mbus/mod.rs

```diff
@@ -0,0 +1,96 @@
+//! Message bus connecting mayastor to the control plane components.
+//!
+//! It is designed to make sending events to control plane easy in the future.
+//!
+//! A Registration subsystem is used to keep moac in the loop
+//! about the lifecycle of mayastor instances.
+
+pub mod mbus_nats;
+pub mod registration;
+
+use async_trait::async_trait;
+use smol::io;
+use spdk_sys::{
+    spdk_subsystem,
+    spdk_subsystem_fini_next,
+    spdk_subsystem_init_next,
+};
+use structopt::StructOpt;
+
+use crate::{
+    core::{MayastorCliArgs, MayastorEnvironment},
+    subsys::mbus::{mbus_nats::NATS_MSG_BUS, registration::Registration},
+};
+
+// wrapper around our MBUS subsystem used for registration
+pub struct MessageBusSubsystem(pub(crate) *mut spdk_subsystem);
```

Comment:
is the suffix "subsystem" needed? IOW MessageBus() would be sufficient?

Reply:
then it conflicts with the actual MessageBus type?

Reply:
Okay, if that is taken, and renaming it does not work then Ok. BTW, the subsystem pointer should ideally not be pub though.

Reply:
hmm yeah that would make sense, maybe the `MessageBusSubsystem` can expose a pub static register_subsystem or something like that.

Reply:
I've exposed it via add which is kept under a generic `add_subsystem` where we could add more subsystems that start after main. Not the best naming I know...